internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -743,11 +735,25 @@ func shouldUpdateIntegrationTestGitResolver(integrationTestScenario *v1beta2.Int
 func getGitResolverUpdateMap(snapshot *applicationapiv1alpha1.Snapshot) map[string]string {
 	annotations := snapshot.GetAnnotations()
 	return map[string]string{
-		tektonconsts.TektonResolverGitParamURL:      urlToGitUrl(annotations[gitops.PipelineAsCodeGitSourceURLAnnotation]), // should have .git in url for consistency and compatibility
+		tektonconsts.TektonResolverGitParamURL:      h.UrlToGitUrl(annotations[gitops.PipelineAsCodeGitSourceURLAnnotation]), // should have .git in url for consistency and compatibility
 		tektonconsts.TektonResolverGitParamRevision: annotations[gitops.PipelineAsCodeSHAAnnotation],
 	}
 }
```

Comment:
Problem:

- No validation that the required annotations exist

- If PipelineAsCodeGitSourceURLAnnotation or PipelineAsCodeSHAAnnotation are missing, this returns empty strings

- This creates invalid PipelineRuns with empty git URLs/revisions

- The PipelineRun creation fails or references wrong/non-existent tasks

Impact: Integration PipelineRuns fail to create or resolve tasks correctly